src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -354,6 +355,32 @@ async def _clone_and_checkout_repositories(
         )
         await _git_fetch(repo.directory)
 
+    # Checking tags
+    if synced_via_tags:
+        # Sanity check
+        at_least_one_repo_has_tag_regex = any(repo.tags for repo in repos)
```

Comment:
e.g. all this "sanity check" you do here can be automatically handled with pydantci settings (i.e. the replacement we are suggesting for the config)

Reply:
this would be nice :)